controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1286,6 +1320,76 @@ bool master_thread::handle_cmdu_1905_link_metric_response(Socket *sd,
     return true;
 }
 
+bool master_thread::handle_cmdu_1905_ap_metric_response(Socket *sd,
+                                                        ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+
+    auto mid = cmdu_rx.getMessageId();
+    LOG(INFO) << "Received AP_METRIC_RESPONSE_MESSAGE, mid=" << std::dec << int(mid);
+
+    //getting reference for ap metric data storage from db
+    auto &ap_metric_data = database.get_ap_metric_data_map(database.get_gw_mac());
+
+    //will hold new ap metric data from Reporting Agent
+    son::node::ap_metrics_data new_ap_metric_data;
```

Comment:
This can be moved inside the loop.

But actually, I don't even think it's needed, we can set it directly in the map without a temporary variable.